akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,18 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case proto ⇒
+        // In OSGi trailing slashes refer to directories. If url already ended with a slash, we wouldn't get here.
+        def isBundleDir = classLoader.getResource(url.getPath + "/") != null
+        if (!proto.equals("bundle") || !isBundleDir) {
+          val conn = url.openConnection()
+          try {
+            conn.setUseCaches(false) // otherwise the JDK will keep the connection open when we close
+            val len = conn.getContentLength
+            val lm = conn.getLastModified
+            Some(ResourceFile(url, len, lm))
+          } finally conn.getInputStream.close() // conn might have already opened the stream
```

Comment:
Thx @sirthias and @jrudolph for unraveling this mystery. :)

This was originally added for `FileURLConnection`s but who knows how other `UrlConnection`s behave.